packages/contracts/src/Game.sol

```diff
@@ -224,19 +224,22 @@ contract Game {
     // =============================================================================================
     // MODIFIERS
 
+
+
+    function _checkIfExists(uint256 _gameId) private {
+        if (gameData[_gameID].lastBlockNum == 0)
+            revert NoGameNoLife();
+    }
+
     // Check that the game exists (has been created and is not finished).
     modifier exists(uint256 gameID) {
-        if (gameData[gameID].lastBlockNum == 0)
-            revert NoGameNoLife();
+        _checkIfExists(gameId);
         _;
```

Comment:
Does this save gas? Using very rough intuition, I'd expect it to only save gas if Solidity doesn't have very basic common sub-expression optimization... which is possible given a rather not great implementation.

(Let's find out! Can you run run both version with make test-gas in the contracts package?)

Even if it does save some gas, I don't think that one is worth the added clutter though.

Reply:
The main motive behind this was not to save gas but to ensure that some core functions are not overloaded and checks like this are forwarded to smaller helper functions and save contract size.
( Overloaded as in : All the modifier's code is inlined into the each function's code where the modifier's being used )

Reply:
Yeah, this was a dumb comment on my part. I got the two things tangled in my head somehow and I was talking about size, but then assuming that it was being used only a single location, which is of course not the case.

Reply:
Still think the size increase due to this is probably not worth making slightly less pleasant to the eye.